include/seqan3/range/container/constexpr_string.hpp

```diff
@@ -0,0 +1,430 @@
+     * No-throw guarantee.
+     */
+    constexpr size_type max_size() const noexcept
+    {
+        return size();
+    }
+
+    /*!\brief Determines whether the string is empty.
+     *
+     * \returns `bool` `true` if `begin() == end()`, `false` otherwise.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr bool empty() const noexcept
+    {
+        return cbegin() == cend();
+    }
+
+    /*!\brief Returns the begin to the string.
+     *
+     * \returns `iterator` An iterator to the begin of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr iterator begin() noexcept
+    {
+        return &_lit[0];
+    }
+
+    /*!\brief Returns the begin to the string.
+     *
+     * \returns `const_iterator` An iterator to the begin of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const_iterator begin() const noexcept
+    {
+        return &_lit[0];
+    }
+
+    /*!\brief Returns the begin to the string.
+     *
+     * \returns `const_iterator` An iterator to the begin of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const_iterator cbegin() const noexcept
+    {
+        return &_lit[0];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `iterator` An iterator to the end of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr iterator end() noexcept
+    {
+        return &_lit[N];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `iterator` An iterator to the end of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const_iterator end() const noexcept
+    {
+        return &_lit[N];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `const_iterator` An iterator to the end of the string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const_iterator cend() const noexcept
+    {
+        return &_lit[N];
+    }
+
+    /*!\brief Compares two strings for equality.
+     *
+     * \returns `bool` `true` if both container are equal, `false` otherwise.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    template <std::size_t N2>
+    constexpr bool operator==(constexpr_string<N2> const & rhs) const noexcept
+    {
+        if constexpr (N != N2)
+            return false;
+
+        const_iterator it_rhs = rhs.cbegin();
+        for (const_iterator it = cbegin(); it != cend(); ++it, ++it_rhs)
+        {
+            if (*it != *it_rhs)
+                return false;
+        }
+
+        return true;
+    }
+
+    /*!\brief Compares two strings for inequality.
+     *
+     * \returns `bool` `false` if both container are equal, `true` otherwise.
+     */
+    template <std::size_t N2>
+    constexpr bool operator!=(constexpr_string<N2> const & rhs) const noexcept
+    {
+        return !(*this == rhs);
+    }
```

Comment:
Both vector and string have the other comparison operators, too. They do lexicographical comparison.
I would also group them in the group "Comparison operators".

Reply:
But this is not necessary for the container concept 😏